src/conn/conn_api.c

```diff
@@ -1822,9 +1822,46 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
               WT_SINGLETHREAD_STRING));
     }
 
+    /*
+     * We own the database home, figure out if we're creating it. There are a few files created when
+     * initializing the database home and we could crash in-between any of them, so there's no
+     * simple test. The last thing we do during initialization is rename a turtle file into place,
+     * and there's never a database home after that point without a turtle file. If the turtle file
+     * doesn't exist, it's a create.
+     */
+    WT_ERR(__wt_turtle_exists(session, &exist));
+    conn->is_new = exist ? 0 : 1;
+
+    /*
+     * If the turtle file exists, the WiredTiger file should exist as well, unless we are salvaging.
+     */
+    WT_ERR(__wt_config_gets(session, cfg, "salvage", &cval));
+    is_salvage = cval.val != 0;
+    if (!is_salvage && !conn->is_new) {
+        WT_ERR(__wt_fs_exist(session, WT_WIREDTIGER, &exist));
+        if (!exist) {
+            F_SET(conn, WT_CONN_DATA_CORRUPTION);
+            WT_ERR_MSG(session, WT_TRY_SALVAGE, "WiredTiger version file cannot be found");
+        }
+    }
+
     /* We own the lock file, optionally create the WiredTiger file. */
-    ret = __wt_open(
-      session, WT_WIREDTIGER, WT_FS_OPEN_FILE_TYPE_REGULAR, is_create ? WT_FS_OPEN_CREATE : 0, &fh);
+    ret = __wt_open(session, WT_WIREDTIGER, WT_FS_OPEN_FILE_TYPE_REGULAR,
+      is_create || is_salvage ? WT_FS_OPEN_CREATE : 0, &fh);
+
+    empty = false;
+    if (fh) {
+        WT_TRET(__wt_filesize(session, fh, &size));
+        if ((size_t)size == 0) {
+            empty = true;
+        }
+        if (!is_salvage && !conn->is_new && empty)
+            /*
+             * If WiredTiger file exists but is size zero when it is not supposed to be (the turtle
+             * file exists and we are not salvaging), write a message but don't fail.
+             */
+            WT_TRET(__wt_msg(session, "WiredTiger version file is empty"));
```

Comment:
Same here, should we fail if the message cannot be printed?

Reply:
I guess it shouldn't fail, since printing of the message is not essential. I have changed this to `WT_IGNORE_RET`. Thanks and let me know what you think.